Deprecate `ZetaSQL`-based `filter_query` functionality for `v1.18.0` removal

czgdp1807 opened the pull request:
## Summary

This PR implements comprehensive deprecation warnings for the `filter_query` parameter in `ListOperationOptions` that depends on ZetaSQL. This prepares for the removal of ZetaSQL dependency from ML Metadata in version 1.18.0.

## Motivation

ZetaSQL dependency is being phased out from ML Metadata. The `filter_query` feature relies on ZetaSQL for SQL query parsing and analysis. To maintain backward compatibility while communicating the planned removal, this PR adds proper deprecation warnings across the codebase.

## Changes

### Runtime Warnings
- **Python API** (`metadata_store.py`): Added `DeprecationWarning` when `filter_query` is used in `ListOptions`
- **C++ Query Executors**: Added `LOG(WARNING)` messages in:
  - `query_config_executor.cc` (MySQL)
  - `postgresql_query_executor.cc` (PostgreSQL)

### Documentation Updates
- **RELEASE.md**: Added deprecation notice in Current Version section with v1.19.0 removal timeline
- **ListOptions docstring**: Updated to clearly mark `filter_query` as deprecated with migration guidance
- **C++ Headers & Implementations**: Added deprecation comments to:
  - `filter_query_ast_resolver.h` and `.cc`
  - `filter_query_builder.h` and `.cc`

### Test Updates
- Modified `test_get_nodes_by_filter_query` to verify `DeprecationWarning` is raised
- Modified `test_get_nodes_by_filter_query_syntax_errors` to verify warnings even on syntax errors
- Added warning suppression (`warnings.catch_warnings()`) to lineage subgraph tests that use `filter_query` for backward compatibility testing
- All deprecation warnings properly tested with assertions on warning message content

## Timeline

- **v1.17.1** (next release): Deprecation warnings introduced
- **v1.18.0**: Complete removal of `filter_query` and ZetaSQL dependency

## Testing

All tests pass with proper deprecation warning handling:
- [x] Tests explicitly verify warnings are raised when `filter_query` is used
- [x] Backward compatibility tests suppress warnings while noting the deprecated functionality
- [x] No functional changes to existing behavior

## Migration Path

Users relying on `filter_query` should migrate to alternative filtering approaches before v1.18.0. The deprecation warnings provide clear guidance on the timeline and necessary actions.

czgdp1807 committed:
feat: deprecate ZetaSQL filter_query for removal in v1.18.0

Deprecate filter_query parameter in ListOperationOptions that depends on
ZetaSQL. Add runtime warnings (Python DeprecationWarning, C++ LOG(WARNING)),
update documentation, and add test coverage for deprecation warnings.

This prepares for removal of ZetaSQL dependency in version 1.18.0.

czgdp1807 committed:
lint: Combine nested with statements into one

czgdp1807 commented:
@aktech @vkarampudi I have updated the title, description and commit messages as per the plan we discussed (to remove ZetaSQL in 1.18.0 release). We should merge this PR before v1.17.1 release (i.e., the next one).

aktech commented on ml_metadata/metadata_store/metadata_store.py:
```diff
@@ -81,9 +81,13 @@ class ListOptions:
-    filter_query: An optional boolean expression in SQL syntax to specify
-      conditions on node attributes and directly connected assets. See
-      https://github.com/google/ml-metadata/blob/master/ml_metadata/proto/metadata_store.proto#L705-L783 for the query capabilities and syntax.
+    filter_query: DEPRECATED (will be removed in v1.19.0) - An optional boolean
```
The PR description states removal in v1.18.0, but comment here says `v1.19.0` and everywhere below.

czgdp1807 replied:
Yes. The plan was changed later. Let me make the updates quickly. I was busy with `struct2tensor`. My 100% focus and time was allocated there (and finally we did it). Now let me clean this PR.

aktech commented:
@czgdp1807 I see a bunch of examples here https://github.com/google/ml-metadata/blob/93574251a56d29f3775dd980daf5dbfbd58840cd/docs/index.md which uses `filter_query` in examples, we need to add note add those places regarding deprecation.

czgdp1807 committed:
Update filter_query deprecation to version 1.18.0

Add deprecation notes to documentation examples and update version references from 1.19.0 to 1.18.0 across Python, C++, and documentation files.

czgdp1807 commented:
This is ready to merge. The tests will pass.

cc: @aktech
